### PR TITLE
[new release] irmin et al. (2.5.0)

### PR DESCRIPTION
**Pull request by CraigFe**
- Project page: <a href="https://github.com/mirage/irmin">https://github.com/mirage/irmin</a>
- Documentation: <a href="https://mirage.github.io/irmin/">https://mirage.github.io/irmin/</a>

### Changed

- **irmin**
  - `Store.Tree.remove` is now much faster when operating on large directories.
    The commits following removals are also much faster. (mirage/irmin#1289, @Ngoguey42)

  - Changed `Store.Tree.{of_hash, shallow}` to take kinded hashes, allowing the
    creation of unforced contents values. (mirage/irmin#1285, @CraigFe)

  - Changed `Tree.destruct` to return _lazy_ contents values, which may be forced
    with `Tree.Contents.force`. (mirage/irmin#1285, @CraigFe)

- **irmin-bench**
  - New features in benchmarks for tree operations (mirage/irmin#1269, @Ngoguey42)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
The tests of irmin-unix failed on arm32. Could you have a look?
```
# ┌───────────────────────────────────────────────────────────────────────────────┐
# │ [FAIL]        GIT                      14   High-level store synchronisation. │
# └───────────────────────────────────────────────────────────────────────────────┘
# +12176884us git-unix.packed-refs [WARNING] Got an error when we tried to load the packed-refs: "(Sys_error \"_build/test-db-git/.git/packed-refs: No such file or directory\")".
# +12177190us git-unix.reference [DEBUG] Writing HEAD: "ref: refs/heads/master\n".
# [exception] Unix.Unix_error(Unix.EAGAIN, "launch_thread", "")
#             Raised by primitive operation at Lwt_unix.run_job_aux in file "src/unix/lwt_unix.cppo.ml", line 183, characters 5-31
#             Called from Lwt.Sequential_composition.catch in file "src/core/lwt.ml", line 2023, characters 16-20
#             Re-raised at Irmin_test__Common.Make_helpers.run.(fun) in file "src/irmin-test/common.ml", line 159, characters 36-72
#             Called from Lwt.Sequential_composition.catch.create_result_promise_and_callback_if_deferred. in file "src/core/lwt.ml", line 2045, characters 23-28
#             Re-raised at Lwt.Miscellaneous.poll in file "src/core/lwt.ml", line 3072, characters 20-29
#             Called from Lwt_main.run.run_loop in file "src/unix/lwt_main.ml", line 31, characters 10-20
#             Called from Lwt_main.run in file "src/unix/lwt_main.ml", line 118, characters 8-13
#             Re-raised at Lwt_main.run in file "src/unix/lwt_main.ml", line 124, characters 4-13
#             Called from Alcotest_engine__Core.Make.protect_test.(fun) in file "src/alcotest-engine/core.ml", line 269, characters 17-23
#             Called from Alcotest_engine__Monad.Identity.catch in file "src/alcotest-engine/monad.ml", line 34, characters 31-35
#             
# 
# Logs saved to `~/.opam/4.11/.opam-switch/build/irmin-unix.2.5.0/_build/default/_build/_tests/irmin-unix/GIT.014.output'.
#  ──────────────────────────────────────────────────────────────────────────────
# 
# Full test results in `~/.opam/4.11/.opam-switch/build/irmin-unix.2.5.0/_build/default/_build/_tests/irmin-unix'.
```

**Comment by CraigFe** [comment]
The test failure is repeatable on the CI and seems to be related to the Lwt bug described in https://github.com/ocsigen/lwt/issues/493.

I don't have much time to look into that right now, so have just added restricted availability of `irmin-unix` for now. Hope that's OK!

**Comment by CraigFe** [comment]
CI failures all appear to be unrelated now.

File: packages/ppx_irmin/ppx_irmin.2.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "dune" {>= "2.7.0"}
+  "ppx_repr" {>= "0.2.0"}
+  "irmin" {with-test & post & = version}
```
Comment (kit-ty-kate) [inline packages/ppx_irmin/ppx_irmin.2.5.0/opam:17]: why is this `post` filter here? If the dependency is not need why is it listed? (same for `irmin-layer`)

Comment (CraigFe) [reply packages/ppx_irmin/ppx_irmin.2.5.0/opam]: I thought we'd finally gotten rid of those :upside_down_face: They are indeed not necessary; I've removed them & done likewise upstream.

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**